### ISSUE TITLE
Failed to get player response for video_id "xxxxxxxxx"

### ISSUE THREAD

**Issue by mediotex** [anxdpanic/plugin.video.youtube#687]
### Context

Please provide any relevant information about your setup

* Add-on Version: YouTube (6.8.25) 
* Kodi Version: Krypton (Kodi-17.6) with Python 2.7.13.
* Kodi GUI Language: Eng
* Operating System: LibreELEC 8.2.3.1
* Operating System Language: Eng

The hardware is Amlogic S905W   ARMv8-A (64-bit), Cortex-A53.

Recently youtube videos in saved playlists are stopped playing, show error: "Failed to get player response for video_id "WgOKPF8LkA8".
I can't upgrade anything, since YouTube 6.8.25 version is the latest available for my hardware. Error logr:

log [example](https://paste.kodi.tv/okixexexeb.kodi)

------

**Comment by MoojMidge** [comment]
The oldest supported version is Kodi 19, and that will soon become unsupported as well.

The plugin will install and may largely work without too many issues on Kodi 18.

As it is unsupported on Kodi 17, you will need to figure things out on your own.

You can modify the addon.xml in the current leia release zip to change the versions for the required dependencies to match the addon.xml in v6.8.25 and it might install. It probably won't run without issues though but the errors may be simple enough for you to fix yourself after doing a search on the internet.


> * Operating System: LibreELEC 8.2.3.1
> The hardware is Amlogic S905W ARMv8-A (64-bit), Cortex-A53.
> I can't upgrade anything, since YouTube 6.8.25 version is the latest available for my hardware.

This is not correct. There are LibreElec builds available for the S905W which use Kodi 18. There are also CoreElec builds which provide Kodi 18/19/20/21 for the S905W.

**Comment by mediotex** [comment]
Thanks for the tip. 
Regarding modding `7.0.5+leia.1` addon.xml to match the addon.xml in `v6.8.25`:  leia `<requires>` section have `<import addon="inputstream.adaptive"/>`  addon which is missing in  `v6.8.25`.

What to do with this?

**Comment by MoojMidge** [comment]
As I mentioned, things won't work properly on Kodi 17, and I can't support using it.

You can try to modify it to force it to install and see what problems may arise, but you are going to have to figure things out on your own and try things out for yourself. If you can't do that then you will be far better off if you update your system. 

Matching the dependencies in `addon.xml` means exactly that, match it to what was in the old version. InputStream.Adaptive is not available on Kodi 17 so it can't be included as a dependency and will need to be removed.

**Comment by mediotex** [comment]
This v6.8.25 add-on worked fine for several years, then stopped. Neither my hardware nor LibreElec OS were updated during this time, so they couldn't affect the operation of this add-on in any way. Therefore I assume this happened due to google changed something in its video streaming algorithm. Probably its possible to localize the issue by changing only the part of the code that's responsible for processing the video stream, without a major upgrades. In particular, are these two scripts, `video_info.py` and `models.py` the most likely reason?

**Comment by MoojMidge** [comment]
> Probably its possible to localize the issue by changing only the part of the code that's responsible for processing the video stream, without a major upgrades. In particular, are these two scripts, `video_info.py` and `models.py` the most likely reason?

Yes, it's possible, depending on your familiarity with Python and motivation to do so. You are welcome to have a go at it and see how it pans out.

I don't know what `models.py` is but you can try and take some of the client detail changes from https://github.com/anxdpanic/plugin.video.youtube/blob/master/resources/lib/youtube_plugin/youtube/client/request_client.py and update the older https://github.com/anxdpanic/plugin.video.youtube/blob/isengard/resources/lib/youtube_plugin/youtube/helper/video_info.py

But like I said, why bother with all of that, just update your system.

**Comment by mediotex** [comment]
Which LibreElec build should I use for S905 update? Is it "Generic Box Image" [LibreELEC-AMLGX.arm-11.0.6-box.img.gz](https://releases.libreelec.tv/LibreELEC-AMLGX.arm-11.0.6-box.img.gz)? 

edit: I tried manually update system, placed above build int /storage/.update  directory, rebooted: it print some update tasks, unpacking image, etc - but finally show same LE version 8.2.3.1, nothing was updated.
